src/pages/index.js

```diff
@@ -1,36 +1,40 @@
 import React from 'react';
 import GatsbyLink from 'gatsby-link';
-import Helmet from 'react-helmet';
+import { graphql } from 'gatsby';
 
 import Link from '../components/Link';
+import Layout from '../components/Layout';
 
 import '../css/index.css';
 
-export default function Index({ data }) {
+export default function Index(props) {
+  const { data } = props;
   const { edges: posts } = data.allMarkdownRemark;
   return (
-    <div className="blog-posts">
-      {posts
-        .filter(post => post.node.frontmatter.title.length > 0)
-        .map(({ node: post }) => {
-          return (
-            <div className="blog-post-preview" key={post.id}>
-              <h1 className="title">
-                <GatsbyLink to={post.frontmatter.path}>
-                  {post.frontmatter.title}
-                </GatsbyLink>
-              </h1>
-              <h2 className="date">
-                {post.frontmatter.date}
-              </h2>
-              <p>
-                {post.excerpt}
-              </p>
-              <Link to={post.frontmatter.path}>Read more</Link>
-            </div>
-          );
-        })}
-    </div>
+    <Layout {...props}>
```

Comment:
Since `Layout` is a regular React component now, I forward all the props that are injected (e.g. location and others) into this Layout component so they come along for the ride.

Reply:
(and this is the only change to remove the magic from the `/layouts/index.js` file!)